### PR TITLE
Fix typo in Arrow optimization config

### PR DESCRIPTION
**Pull request by andygrove**
Signed-off-by: Andy Grove <andygrove@nvidia.com>

This typo was fixed in the past but seems to have reverted in a subsequent PR merge issue.

### CODE REVIEW

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuExpressions.scala [hunk 1]
```diff
@@ -194,7 +194,7 @@ trait CudfBinaryExpression extends GpuBinaryExpression {
   def outputType(l: BinaryOperable, r: BinaryOperable) : DType = {
     val over = outputTypeOverride
     if (over == null) {
-      BinaryOperable.implicitConversion(l, r)
+      BinaryOperable.implicitConversion(binaryOp, l, r)
```
Comment (jlowe) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuExpressions.scala:197]: This seems out of place in a PR to fix a typo.  Probably best to wait for https://github.com/NVIDIA/spark-rapids/pull/1889 to be merged and upmerge afterwards to pass premerge CI.

**Comment by andygrove** [comment]
build

**Review by revans2 — COMMENTED** [review]
I am fine with fixing the typo, it is on by default and internal so I don't suspect that anyone will actually use this config in the 0.4 release, otherwise we will have to add in support for deprecating/remapping a config.

**Review by revans2 — APPROVED** [review]

**Comment by tgravescs** [comment]
this is weird, I fixed this already, did it not go into 0.5 - https://github.com/NVIDIA/spark-rapids/pull/1652/files

**Comment by tgravescs** [comment]
oh sorry, just saw comment about it being reverted

**Pull request merged**